IdentityCore/src/util/NSString+MSIDExtensions.m

```diff
@@ -165,7 +165,7 @@ + (NSString *)msidHexStringFromData:(NSData *)data
     NSUInteger dataLength = data.length;
     NSMutableString *result = [NSMutableString stringWithCapacity:dataLength*2];
     
-    for (int i = 0; i < dataLength; i++)
+    for (unsigned int i = 0; i < dataLength; i++)
```

Comment:
NSUInteger can also be long, wouldn't using "unsigned int" also result in a warning on some machines?
Btw, we don't see that warning in ADAL/MSAL, can you share your settings with us?

Reply:
The main thing is we build our project with and `-Werror` (treat warnings as errors) and a lot of warnings enabled, including `-Wsign-compare`:

`/Users/dean/repos/msalobjc/IdentityCore/src/util/NSString+MSIDExtensions.m:168:23: error: comparison of integers of different signs: 'int' and 'NSUInteger' (aka 'unsigned long') [-Werror,-Wsign-compare]`

A better fix would be declaring `i` as `NSUInteger` so it always matches `dataLength`'s type exactly.